### DISCUSSION TITLE
Implementing graph with axis independent zoom

### DISCUSSION THREAD

**Discussion by MichalLebeda** [hecrj/iced#1460] [Q&A] [answered]
Hi,
how can I implement a graph with a zoom feature? I'm using Canvas but `Frame::scale` takes **only one argument** and **scales both axes uniformly**. It would be great to be able to **be able to scale both axes interdependently** or am I missing something?

**Comment by 13r0ck** [comment, marked as answer]
This was answered by @tarkah on the iced discord

https://discord.com/channels/628993209984614400/1027684406388863068/1027686061897429002

```
You should handle this with your own conversion between input domain / pixel domain.

That way you can define a 2:1 scale on y and 1:1 on x or however you want it. Then you can convert both ways.
```
